Client/Application/AppDelegate.swift

```diff
@@ -192,7 +192,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             // Ecosia: Lifecycle tracking. Needs to happen after Unleash start so that the flags are correctly added to the analytics context.
             Analytics.shared.activity(.launch)
             // Ecosia: Engagement Service Initialization helper
-            await ClientEngagementService.shared.initializeAndRefreshNotificationRegistration(notificationCenterDelegate: self)
```

Comment:
The `ClientEngagementService` (which could also be named `EngagementService` now) was created to "wrap" the explicit implementation of `Braze` behind a generic service.  
Could that be achieved anyway or is there an explicit reason for renaming it into `BrazeService`?

Reply:
I understand the desire to abstract this provider layer, though in my opinion it was already becoming very Braze specific and that was I though it would be better to simplify it. Should be easy enough to add this `EngagementService` layer again though if you prefer. Do you see any advantages for that even if we don't foresee changing providers in the future?